### PR TITLE
fix(savesets): #3233 fix participant saved sets bug

### PR DESCRIPTION
**Pull request by oliviercperrier**

### CODE REVIEW

**Comment by oliviercperrier** [comment]
Did some refactoring at the same time to use hooks in participantTableView

**Review by celinepelletier — APPROVED** [review]

File: src/components/CohortBuilder/ParticipantsTableView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            <div className="ptv-error-message">
+              <TableErrorView error={error} />
+            </div>
+          );
+        }
+
+        const isRowSelected = (node: any) =>
+          allRowsSelected || selectedRows.some((row) => row === node.participantId);
+
+        const dataWithRowSelection = data[0]
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/ParticipantsTableView/index.tsx:59]: a?.b?.c || []

File: src/components/CohortBuilder/ParticipantsTableView/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          : [];
+
+        return (
+          <ParticipantsTable
+            sqon={props.sqon}
+            loading={isLoading}
+            data={dataWithRowSelection}
+            api={props.api}
+            sort={sort}
+            dataTotalCount={data[0] ? data[0].total : 0}
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/ParticipantsTableView/index.tsx:70]: data[0]?total || 0

File: src/components/CohortBuilder/ParticipantsTableView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            downloadName={'participant-table'}
+            onFetchData={({
+              page,
+              pageSize,
+              sorted,
+            }: {
+              page: any;
```
Comment (evans-g-crsj) [inline src/components/CohortBuilder/ParticipantsTableView/index.tsx:77]: no `any`'s unless you can't do otherwise

**Review by evans-g-crsj — CHANGES_REQUESTED** [review]

File: src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.tsx [hunk 4]
```diff
@@ -90,13 +98,49 @@ const ParticipantSetDropdown = ({
...
-    <Menu onClick={onClick}>
+    <Menu className="save-set-option-menu" onClick={onClick}>
+      <Menu.Item
+        id="participant-count"
+        key="participant-count"
+        className={exceedLimit() ? 'over' : ''}
+        disabled
+        icon={
+          <DemographicIcon
+            fill={exceedLimit() ? '#dd1f2a' : '#a9adc0'}
```
[the remark this answers is not in the document]
Comment (oliviercperrier) [reply src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.tsx:128]: Its a component rendering an SVG and i'm only using the props exposed. I can't really use a css classe here.

File: src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.scss [hunk 5]
```diff
@@ -0,0 +1,51 @@
+@import 'style/themes/default/_colors';
+
+.ant-dropdown-menu-item:not(.ant-dropdown-menu-item-disabled) {
```
[the remark this answers is not in the document]
Comment (oliviercperrier) [reply src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.scss:3]: Was already there, but i can fix it

File: src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (evans-g-crsj) [reply src/components/CohortBuilder/ParticipantsTableView/ParticipantSetDropdown.tsx:130]: no probs

File: src/components/CohortBuilder/Results.js [hunk 6]
```diff
@@ -60,84 +60,84 @@ const cohortResultsQuery = (sqon) => ({
...
+        activeSqonIndex,
+        sqon,
       }}
-    </QueriesResolver>
+    />
+  );
+
+  return (
+    <>
+      <div style={{ padding: '0 30px 0 34px' }} className="cb-view-links">
```
Comment (celinepelletier) [inline src/components/CohortBuilder/Results.js:94]: why do you put padding in style here instead of putting in a css class ?

**Review by celinepelletier — APPROVED** [review]

**Pull request merged**